_posts/2019-06-24-logging-call-into-your-service-with-ringcentral-embeddable.md

```diff
@@ -0,0 +1,153 @@
+  })();
+</script>
+```
+
+The script will create a RingCentral widget into your website, so your customer can login with RingCentral account to access RingCentral services:
+
+![RingCentral Embeddable widget](https://ringcentral-web-widget-demos.readthedocs.io/en/latest/static_crm/tutorial/static_crm_demo.png)
+
+## Register call logging service into Embeddable
+
+The `Adapter JS` will create a iframe in webpage to embed RingCentral Embeddable. And we can interact with RingCentral Embeddable by iframe message API.
+
+Firstly, we need to register call logging service into Embeddable:
+
+```js
+document.querySelector("#rc-widget-adapter-frame").contentWindow.postMessage({
+  type: 'rc-adapter-register-third-party-service',
+  service: {
+    name: 'TestService',
+    callLoggerPath: '/callLogger',
+    callLoggerTitle: 'Log to TestService',
+  }
+}, '*');
+```
+
+After registered, we can get a `Log to TestService` button in calls page, and `Auto log calls` setting in settings page:
+
+![RingCentral Embeddable log to service](https://user-images.githubusercontent.com/7036536/48827686-d1814a00-eda8-11e8-81e4-2b48b1df2bcc.png)
+
+## Respond to RingCentral Embeddable's event message
+
+When user clicks `Log` button in calls page, the widget will send a message including call info to its parent window. So we just need to add a event listener:
+
+```js
+window.addEventListener('message', function (e) {
+  var data = e.data;
+  if (data && data.type === 'rc-post-message-request') {
+    if (data.path === '/callLogger') {
+      // add your codes here to log call to your service
+      console.log(data);
+      // response to widget
+      document.querySelector("#rc-widget-adapter-frame").contentWindow.postMessage({
+        type: 'rc-post-message-response',
+        responseId: data.requestId,
+        response: { data: 'ok' },
+      }, '*');
+    }
+  }
+```
+
+If user enables `Auto log calls` in settings, this event will be also fired when a call is started and updated. In this message event, you can get call information in `data.body.call`. When call is recorded and recording file is generated, you can get recording data in `data.body.call`:
+
+```js
+{
+  contentUri: "https://media.devtest.ringcentral.com/restapi/v1.0/account/170848004/recording/6469338004/content"
+  id: "6469338004"
+  link: "http://apps.ringcentral.com/integrations/recording/sandbox/?id=Ab7937-59r6EzUA&recordingId=6469338004"
+  type: "OnDemand"
+  uri: "https://platform.devtest.ringcentral.com/restapi/v1.0/account/170848004/recording/6469338004"
+}
+```
+
+The `link` property in `recording` is a link to get and play recording file from RingCentral server. The `contentUri` is a URI which can be used to get `recording` file with RingCentral access token. If you pass `recordingWithToken` when register service, you can get contentUri with access_token. The access_token will be expired in minutes, so need to download immediately when get it.
```

Comment:
so need to download immediately when get it => so `you` need to download immediately `after getting` it.